src/libraries/System.Net.Http/ref/System.Net.Http.cs

```diff
@@ -505,6 +522,20 @@ public abstract partial class HttpHeaders : System.Collections.Generic.IEnumerab
         public bool TryAddWithoutValidation(string name, string? value) { throw null; }
         public bool TryGetValues(string name, [System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] out System.Collections.Generic.IEnumerable<string>? values) { throw null; }
     }
+    public readonly partial struct HttpHeadersNonValidated : System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Net.Http.Headers.HeaderStringValues>>, System.Collections.IEnumerable
+    {
+        public System.Collections.Generic.IEnumerator<System.Collections.Generic.KeyValuePair<string, System.Net.Http.Headers.HeaderStringValues>> GetEnumerator() { throw null; }
```

Comment:
This needs to be `Enumerator`, not `System.Collections.Generic.IEnumerator<System.Collections.Generic.KeyValuePair<string, System.Net.Http.Headers.HeaderStringValues>>`

Reply:
I can change it of course, but it was done the same way [as HttpHeaders.GetEnumerator](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/ref/System.Net.Http.cs#L515), so I thought it's intentional. No?

```C#
public System.Collections.Generic.IEnumerator<System.Collections.Generic.KeyValuePair<string, System.Collections.Generic.IEnumerable<string>>> GetEnumerator() { throw null; }
```

Reply:
[HttpHeaders.GetEnumerator returns IEnumerator](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs#L308) because if `_headerStore` is empty it returns Array.Empty<>.Enumerator.

Reply:
The whole point of having the struct enumerator is to avoid the allocation.  This needs to change.  HttpHeaders.GetEnumerator uses Array.Empty's enumerator as an implementation detail _because_ the API was defined to return the interface.  If it had been defined to return a struct, it would have been implemented differently.

Reply:
Ok. That's clear, then.

Reply:
Done.